e2e tests added

xbabalov opened the pull request:
Added e2e tests that cover extension installation and handling.
Issues: https://github.com/redhat-developer/podman-desktop-sandbox-ext/issues/199 and https://github.com/redhat-developer/podman-desktop-sandbox-ext/issues/201

jeffmaury commented on package.json:
```diff
@@ -61,29 +62,37 @@
     "got": "^14.4.2",
-    "js-yaml": "^4.1.0"
+    "js-yaml": "^4.1.0",
+    "node-fetch": "^2.6.1"
```
Should be in devDependencies ?

jeffmaury reviewed (changes requested):
Also needs rebase

odockal commented on vitest.config.js:
```diff
@@ -26,6 +26,11 @@ const config = {
     globals: true,
     coverage: {
       include: ['**/*.ts'],
+      exclude: ['tests/**', '**/builtin/**',
```
this is coverage part. We should not touch coverage, we might add `tests` folder though here.

odockal commented on vitest.config.js:
```diff
@@ -26,6 +26,11 @@ const config = {
     globals: true,
     coverage: {
       include: ['**/*.ts'],
```
WE may explicitly include unit tests (omitting e2e tests), see here: https://github.com/redhat-developer/podman-desktop-redhat-account-ext/blob/main/vitest.config.js#L66

odockal commented on package.json:
```diff
@@ -61,29 +62,37 @@
     "watch": "vite build -w",
     "format:check": "prettier --check \"**/*.ts\" \"scripts/*.js\"",
```
if you change name of the script `build.js` -> `build.cjs` it also needs to be reflected here.

odockal commented on tests/src/developer-sandbox.spec.ts:
```diff
@@ -0,0 +1,184 @@
+    // we want to install extension from OCI image (usually using latest tag) after new code was added to the codebase
+    // and extension was published already
+    test('Extension can be installed using OCI image', async ({ navigationBar }) => {
+      test.skip(extensionInstalled && !skipInstallation);
```
this condition needs to be rewritten into: `!!skipInstallation` otherwise it does not work as expected in PR check.

odockal reviewed (changes requested):
Quite ok, except the review comments. but e2e tests are overal ok!

jeffmaury reviewed (changes requested):
Such a PR should be split in 2 parts: switch to module then add E2E tests

odockal commented:
@jeffmaury @xbabalov First we need to switch to ES modules and then we could merge this PR. I will open an issue.

odockal commented:
@xbabalov please open a PR to fix https://github.com/redhat-developer/podman-desktop-sandbox-ext/issues/219 which changes related to switch to ES modules.

xbabalov commented:
https://github.com/redhat-developer/podman-desktop-sandbox-ext/pull/221 was merged, I rebased and updated the files, so this should be ready for merge as well.

odockal reviewed (approved):
LGTM

jeffmaury commented on package.json:
```diff
@@ -62,29 +62,37 @@
+    "xvfb-maybe": "^0.2.1",
+    "zip-local": "^0.3.5",
+    "js-yaml": "^4.1.0",
+    "node-fetch": "^2.6.1"
```
node-fetch is to be removed as fetch is now native since fetch is now part of nodejs

jeffmaury commented on package.json:
```diff
@@ -62,29 +62,36 @@
     "fs-extra": "^11.2.0",
-    "got": "^14.4.3",
-    "js-yaml": "^4.1.0"
+    "got": "^14.4.2"
```
Why did got version get lowered ?

jeffmaury commented on package.json:
```diff
@@ -62,29 +62,36 @@
   "devDependencies": {
-    "7zip-min": "^1.4.5",
+    "@playwright/test": "1.48.0",
+    "@podman-desktop/tests-playwright": "next",
```
Not sure we should use next as it is unreliable

odockal replied:
I think it is ok. We are replacing it with latest in the CI. Otherwise it gets latest stable released version of the package in lock file.

jeffmaury reviewed (approved):
LGTM

odockal commented:
@xbabalov @jeffmaury Thanks!
